packages/contracts/contracts/MarketLiquidityRewards.sol

```diff
@@ -0,0 +1,474 @@
+    }
+
+    /**
+     * @notice Allows anyone to add tokens to an allocation
+     * @param _allocationId - The id for the allocation
+     * @param _tokenAmount - The amount of tokens to add
+     */
+    function increaseAllocationAmount(
+        uint256 _allocationId,
+        uint256 _tokenAmount
+    ) public virtual {
+        IERC20Upgradeable(allocatedRewards[_allocationId].rewardTokenAddress)
+            .transferFrom(msg.sender, address(this), _tokenAmount);
+        allocatedRewards[_allocationId].rewardTokenAmount += _tokenAmount;
+
+        emit IncreasedAllocation(_allocationId, _tokenAmount);
+    }
+
+    /**
+     * @notice Allows the allocator to withdraw some of all of the funds within an allocation
+     * @param _allocationId - The id for the allocation
+     * @param _tokenAmount - The amount of tokens to withdraw
+     */
+    function deallocateRewards(uint256 _allocationId, uint256 _tokenAmount)
+        public
+        virtual
+    {
+        require(
+            msg.sender == allocatedRewards[_allocationId].allocator,
+            "Only the allocator can deallocate rewards."
+        );
+
+        //subtract amount reward before transfer
+        allocatedRewards[_allocationId].rewardTokenAmount -= _tokenAmount;
+
+        IERC20Upgradeable(allocatedRewards[_allocationId].rewardTokenAddress)
+            .transfer(msg.sender, _tokenAmount);
+
+        emit DecreasedAllocation(_allocationId, _tokenAmount);
+
+        if (allocatedRewards[_allocationId].rewardTokenAmount == 0) {
+            delete allocatedRewards[_allocationId];
+
+            emit DeletedAllocation(_allocationId);
+        }
+    }
+
+    /**
+     * @notice Allows a borrower or lender to withdraw the allocated ERC20 reward for their loan
+     * @param _allocationId - The id for the reward allocation
+     * @param _bidId - The id for the loan.  Each loan only grants one reward per allocation.
+     */
+    function claimRewards(uint256 _allocationId, uint256 _bidId)
+        external
+        virtual
+    {
+        RewardAllocation storage allocatedReward = allocatedRewards[
+            _allocationId
+        ];
+
+        require(
+            !rewardClaimedForBid[_bidId][_allocationId],
+            "reward already claimed"
+        );
+        rewardClaimedForBid[_bidId][_allocationId] = true; // leave this here to defend against re-entrancy
+
+        //optimize gas by turning these into one single call
+
+        (
+            address borrower,
+            address lender,
+            uint256 marketId,
+            address principalTokenAddress,
+            uint256 principalAmount,
+            uint32 timestamp,
+            BidState bidState
+        ) = ITellerV2(tellerV2).getLoanSummary(_bidId);
+
+        address collateralTokenAddress = allocatedReward
+            .requiredCollateralTokenAddress;
+
+        //require that the loan was started in the correct timeframe
+        _verifyLoanStartTime(
+            timestamp,
```

Comment:
we want the `acceeptedTimestamp` here - `timestamp` on the bid is when the bid was submitted

Reply:
accepted timestamp is being used now

Reply:
rename to acceptedtimestamp